torchvision/transforms/v2/functional/_geometry.py

```diff
@@ -585,13 +585,8 @@ def _apply_grid_transform(img: torch.Tensor, grid: torch.Tensor, mode: str, fill
         mask = mask.expand_as(float_img)
         fill_list = fill if isinstance(fill, (tuple, list)) else [float(fill)]  # type: ignore[arg-type]
         fill_img = torch.tensor(fill_list, dtype=float_img.dtype, device=float_img.device).view(1, -1, 1, 1)
-        if mode == "nearest":
-            bool_mask = mask < 0.5
-            float_img[bool_mask] = fill_img.expand_as(float_img)[bool_mask]
-        else:  # 'bilinear'
-            # The following is mathematically equivalent to:
-            # img * mask + (1.0 - mask) * fill = img * mask - fill * mask + fill = mask * (img - fill) + fill
-            float_img = float_img.sub_(fill_img).mul_(mask).add_(fill_img)
+        bool_mask = mask < 1
+        float_img[bool_mask] = fill_img.expand_as(float_img)[bool_mask]
```

Comment:
This is only equivalent to the 'bilinear' block if `mask` was a boolean `mask`, right?

Is that the case? And if it is, why was there even a distinction between bilinear and nearest mode in the first place?

Reply:
> This is only equivalent to the 'bilinear' block if `mask` was a boolean `mask`, right?

Yes.

> Is that the case?

No. 

> And if it is, why was there even a distinction between bilinear and nearest mode in the first place?

I would need to dig, but maybe @vfdev-5 knows? My guess is that the blend strategy we implemented was to have a smooth transition from fill to image. And that works well if we fill with zeros, but creates a shadow artifact if we use a different fill color with bilinear interpolation. See #8083.

Meaning, this PR is BC breaking, but I consider it a bug fix since the blending behavior seems to be "nice to have" while the shadow is an actual issue.

Reply:
I think it was an incorrect implementation. At the moment of PR (https://github.com/pytorch/vision/pull/2904), I thought it could be ok as assumption to linearly interpolate around the boundaries: https://github.com/pytorch/vision/pull/2904#issuecomment-720548451